internal/runner/jest.go

```diff
@@ -0,0 +1,184 @@
+		if err != nil {
+			return RunResult{Status: RunStatusError}, fmt.Errorf("failed to build command: %w", err)
+		}
+
+		fmt.Printf("%s %s\n", commandName, strings.Join(commandArgs, " "))
+		cmd = exec.Command(commandName, commandArgs...)
+	} else {
+		commandName, commandArgs, err := j.retryCommandNameAndArgs(j.RetryTestCommand, testCases)
+		if err != nil {
+			return RunResult{Status: RunStatusError}, fmt.Errorf("failed to build command: %w", err)
+		}
+
+		fmt.Println(commandName)
+		fmt.Println(commandArgs)
+
+		cmd = exec.Command(commandName, commandArgs...)
+	}
+
+	err = runAndForwardSignal(cmd)
+
+	if err == nil {
+		return RunResult{Status: RunStatusPassed}, nil
+	}
+
+	if ProcessSignaledError := new(ProcessSignaledError); errors.As(err, &ProcessSignaledError) {
+		return RunResult{Status: RunStatusError}, err
+	}
+
+	if exitError := new(exec.ExitError); errors.As(err, &exitError) {
+		report, parseErr := j.ParseReport("tmp/jest.json")
+		if parseErr != nil {
+			fmt.Println("Buildkite Test Splitter: Failed to read Jest output, tests will not be retried.")
+			return RunResult{Status: RunStatusError}, err
+		}
+
+		if report.NumFailedTests > 0 {
+			var failedTests []string
+			for _, testResult := range report.TestResults {
+				for _, example := range testResult.AssertionResults {
+					if example.Status == "failed" {
+						failedTests = append(failedTests, example.Name)
+					}
+				}
+			}
+
+			return RunResult{Status: RunStatusFailed, FailedTests: failedTests}, nil
+		}
+	}
+
+	return RunResult{Status: RunStatusError}, err
+}
+
+type JestExample struct {
+	Name   string `json:"fullName"`
+	Status string `json:"status"`
+}
+
+type JestReport struct {
+	NumFailedTests int `json:"numFailedTests"`
+	TestResults    []struct {
+		AssertionResults []JestExample
+	}
+}
+
+func (j Jest) ParseReport(path string) (JestReport, error) {
+	var report JestReport
+	data, err := os.ReadFile(path)
+	if err != nil {
+		return JestReport{}, fmt.Errorf("failed to read Jest output: %v", err)
+	}
+
+	if err := json.Unmarshal(data, &report); err != nil {
+		return JestReport{}, fmt.Errorf("failed to parse Jest output: %s", err)
+	}
+
+	return report, nil
+}
+
+func (j Jest) commandNameAndArgs(cmd string, testCases []string) (string, []string, error) {
+	words, err := shellquote.Split(cmd)
+	if err != nil {
+		return "", []string{}, err
+	}
+	idx := slices.Index(words, "{{testExamples}}")
+	if idx < 0 {
+		words = append(words, testCases...)
+		return words[0], words[1:], nil
+	}
+	words = slices.Replace(words, idx, idx+1, testCases...)
+	return words[0], words[1:], nil
+}
+
+func (j Jest) retryCommandNameAndArgs(cmd string, testCases []string) (string, []string, error) {
+	words, err := shellquote.Split(cmd)
+	if err != nil {
+		return "", []string{}, err
+	}
+
+	idx := slices.Index(words, "{{testNamePattern}}")
+	if idx < 0 {
+		err := fmt.Errorf("Couldn't find '{{testNamePattern}}' sentinel in retry command, exiting.")
```

Comment:
```suggestion
		err := fmt.Errorf("couldn't find '{{testNamePattern}}' sentinel in retry command")
```

[Error string styleguide](https://google.github.io/styleguide/go/decisions.html#error-strings)

In convention, error string in Golang should not be capitalized. Also, I think `exiting` message should be logged from the consumer (main fn), as it is the responsibility of consumer to exit the program.